internal/app/mymemory/mymemory.go

```diff
@@ -0,0 +1,69 @@
+package mymemory
+
+import (
+	"crypto/rand"
+	"fmt"
+)
+
+type MyMemory struct {
+	myMemory map[string]string
+}
+
+func InitMyMemory(initValue map[string]string) *MyMemory {
+	m := MyMemory{}
+	m.myMemory = initValue
+
+	return &m
+}
+
+func (m *MyMemory) CreateOrGetFromMemory(url []byte) (string, error) {
+	val, ok := m.myMemory[string(url)]
+	// If the key exists
+	if ok {
+		return val, nil
+	}
+	// input into our memory
+	if val, err := createShortURL(url); err == nil {
+		m.myMemory[string(url)] = val
+		return val, nil
+	} else {
+		return "", err
+	}
+}
+
+func (m MyMemory) GetFromMemory(url []byte) (string, error) {
```

Comment:
Why don't you use a pointer here?

Reply:
By test i sure that this function with passing struct working more quikliy rather than with passing pointer and func not going to change a struct.
This structure not so big to pass pointer.
If i will work with structure more bigger - i will refactor it by passing pointer.